openstack/utils/templates/_hosts.tpl

```diff
@@ -23,7 +23,13 @@ postgresql+psycopg2://{{$user}}:{{$password | urlquery}}@{{.Chart.Name}}-postgre
 
 {{- define "db_credentials" }}
     {{- if kindIs "map" . }}
-        {{- coalesce .Values.dbUser .Values.global.dbUser "root" }}:{{ coalesce .Values.dbPassword .Values.global.dbPassword .Values.mariadb.root_password | required ".Values.mariadb.root_password is required!" | urlquery }}
+        {{- if and .Values.mariadb.users .Values.mariadb.databases }}
+            {{- $db := first .Values.mariadb.databases }}
+            {{- $user := get .Values.mariadb.users $db | required (print "User with key of the first database in .Values.mariadb.databases (" $db ") required") }}
+            {{- $user.name | required (print ".Values.mariadb.%v.name is required!" $db ) }}:{{ $user.password | required (print ".Values.mariadb.%v.password is required!" $db ) }}
```

Comment:
I think this need to be `printf` instead of `print`.

Reply:
True, fixed it.